window-manager/src/androidTest/java/com/codelab/foldables/window_manager/MainActivityTest.kt

```diff
@@ -0,0 +1,113 @@
+import androidx.window.testing.layout.WindowLayoutInfoPublisherRule
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.async
+import kotlinx.coroutines.flow.first
+import kotlinx.coroutines.test.TestCoroutineScope
+import kotlinx.coroutines.test.runBlockingTest
+import org.junit.Assert.assertEquals
+import org.junit.Rule
+import org.junit.Test
+import org.junit.rules.RuleChain
+import org.junit.rules.TestRule
+import org.junit.runner.RunWith
+
+@OptIn(ExperimentalCoroutinesApi::class)
+@RunWith(AndroidJUnit4::class)
+class MainActivityTest {
+    private val activityRule = ActivityScenarioRule(MainActivity::class.java)
+    private val publisherRule = WindowLayoutInfoPublisherRule()
+
+    private val testScope = TestCoroutineScope()
+
+    @get:Rule
+    val testRule: TestRule
+
+    init {
+        testRule = RuleChain.outerRule(publisherRule).around(activityRule)
+    }
+
+    @Test
+    fun testText_is_left_of_Vertical_FoldingFeature() = runBlockingTest {
+        activityRule.scenario.onActivity { activity ->
+            val hinge = FoldingFeature(
+                activity = activity,
+                state = FoldingFeature.State.FLAT,
+                orientation = FoldingFeature.Orientation.VERTICAL,
+                size = 2
+            )
+            val expected =
+                WindowLayoutInfo.Builder().setDisplayFeatures(listOf(hinge)).build()
+
+            val value = testScope.async {
+                activity.windowInfoRepository().windowLayoutInfo.first()
+            }
+            publisherRule.overrideWindowLayoutInfo(expected)
+            runBlockingTest {
+                assertEquals(
+                    expected,
+                    value.await()
+                )
+            }
+        }
+        onView(withId(R.id.layout_change)).check(
+            PositionAssertions.isCompletelyLeftOf(withId(R.id.folding_feature))
+        )
+    }
+
+    @Test
+    fun testText_is_below_of_Horizontal_FoldingFeature() = runBlockingTest {
+        activityRule.scenario.onActivity { activity ->
+            val hinge = FoldingFeature(
+                activity = activity,
+                state = FoldingFeature.State.FLAT,
+                orientation = FoldingFeature.Orientation.HORIZONTAL,
+                size = 2
+            )
+            val expected =
+                WindowLayoutInfo.Builder().setDisplayFeatures(listOf(hinge)).build()
+
+            val value = testScope.async {
+                activity.windowInfoRepository().windowLayoutInfo.first()
+            }
+            publisherRule.overrideWindowLayoutInfo(expected)
+            runBlockingTest {
+                assertEquals(
+                    expected,
+                    value.await()
+                )
+            }
+        }
+        onView(withId(R.id.layout_change)).check(
+            PositionAssertions.isCompletelyBelow(withId(R.id.folding_feature))
+        )
+    }
+}
```

Comment:
nit: missing

Reply:
Addressed in new commits. Thanks Pietro!